### PR TITLE
Lecture 05 Notes

### PR DESCRIPTION
**Pull request by cphalen**

### CODE REVIEW

File: content/lectures/lec05.md [hunk 1]
```diff
@@ -3,8 +3,149 @@ title: "Practical Kubernetes"
...
+![DaemonSet Diagram][/img/lec05/daemonset.png]
+
+## Annotations
+
+[Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) aren't resources on their own, but a type of resource metadata used to store auxillary key/value information about a resource. Annotations aren't used by Kubernetes itself, but other software in and around your cluster might use the extra metadata to determine its own behavior. 
+
+## Kubernetes as an Abstraction
+
+(@peyton or @armaan can you rewrite this paragraph? not sure i'm gonna phrase it better than y'all)
+Sometimes instead of thinking about Kubernetes as a container orchestration tool, it can be better to think about it as an abstraction that allows us to reason about infrastructure. Kubernetes is the most popular container orchestration tool not because it is the best at orchestrating containers, but because the abstractions are simple and elegant.
```
Comment (pawalt) [inline content/lectures/lec05.md:151]: i'll rewrite along with the demo pass. just move the note into a markdown comment for now

File: content/lectures/lec05.md [hunk 2]
```diff
@@ -3,8 +3,149 @@ title: "Practical Kubernetes"
...
+
+## DaemonSet
+
+A [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) is a deployment which has one pod running on each node in the cluster. These are used to provide node-wide services like authentication gateways, ingress routers, and log collectors.
+
+![DaemonSet Diagram][/img/lec05/daemonset.png]
+
+## Annotations
+
+[Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) aren't resources on their own, but a type of resource metadata used to store auxillary key/value information about a resource. Annotations aren't used by Kubernetes itself, but other software in and around your cluster might use the extra metadata to determine its own behavior.
```
Comment (pawalt) [inline content/lectures/lec05.md:146]: Annotations are used by Kubernetes. For example, kubectl stores its last applied config in an annotation. The key thing to realize about labels vs annotations is that annotations are non-identifying i.e. they're only used for storing metadata, not for doing resource lookups.

**Review by pawalt — CHANGES_REQUESTED** [review]

File: content/lectures/lec05.md [hunk 3]
```diff
@@ -3,8 +3,149 @@ title: "Practical Kubernetes"
...
+            imagePullPolicy: IfNotPresent
+            args:
+            - /bin/sh
+            - -c
+            - date; echo Hello from the Kubernetes cluster
+          restartPolicy: OnFailure
+
+```
+
+Why do we have to nest four or five YAML objects with `template` and `spec` just to run a job on a schedule? The answer comes down to modularity and the construction of the API. A CronJob is just a Job with a schedule, and a Job is in turn just a Pod that runs to completion. You can think of it as a very rough form of inheritance in Java. Kubernetes takes advantage of this conceptual hierarchy, and embeds references to API objects within API objects. In a quite literal sense, a CronJob has a Job, which in turn has a Pod. While this modularity results in more complex YAML, it makes it much easier for us to reason about resources which are simply compositions of other resources.
```
[the remark this answers is not in the document]
Comment (davish) [reply content/lectures/lec05.md:117]: I knew this wasn't right lol sorry for sneaking this in

File: content/lectures/lec05.md (hunk 2 above)
Comment (pawalt) [reply content/lectures/lec05.md:146]: Also an example of an annotation here would be helpful

File: content/lectures/lec05.md [hunk 4]
```diff
@@ -3,8 +3,149 @@ title: "Practical Kubernetes"
...
+[Namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) are a way to separate resources by a common name. They're useful for organization as well as permissioning. While most Kubernetes resources are scoped to a namespace, resources having to do with the whole cluster are not (e.g. ClusterRoleBindings). Namespaces aren't a foolproof way to separate resources within a cluster, because they don't provide any strong security guarantees. 
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cphalen
+```
+
+Two namespaces are available by default in a cluster: `default` for your own resources and `kube-system`, for resources the cluster needs to run itself. You can get resources belonging to a certain cluster with by adding `-n kube-system` to a `kubectl get` command.
```
[the remark this answers is not in the document]
Comment (davish) [reply content/lectures/lec05.md:25]: sorry, meant namespace. edited.

File: content/lectures/lec05.md (hunk 2 above)
Comment (davish) [reply content/lectures/lec05.md:146]: doesn’t kubernetes itself use the status field on resources to handle things, rather than annotations?

**Pull request merged**